### PR TITLE
Auth api call

### PR DESCRIPTION
**Pull request by rutvik-balar**
**Task**
Auth api call
 
**Todo**
 
add api call of login API 
add api call of register and valid otp send otp api in register screen
 
**How to check**
npm run dev

**Evidences**
 
[screen-capture (1).webm](https://github.com/r-aritra/outdoor-event-fe/assets/76244286/ba5b4e81-63d4-4f72-bd5b-6bd9788e6976)

 
**build**
**Related Links**

### CODE REVIEW

File: src/features/Login/Login.tsx [hunk 1]
```diff
@@ -4,26 +4,26 @@ import {
...
   const form = useForm({
     initialValues: {
-      email: '',
-      name: '',
+      email: state ? state.email : '',
       password: '',
-      terms: true,
     },
 
     validate: {
```
Comment (r-aritra) [inline src/features/Login/Login.tsx:29]: use zod for validation schema

File: src/features/Login/Login.tsx [hunk 2]
```diff
@@ -33,21 +33,41 @@ export default function Login() {
     },
   });
 
+  const showNotification = (title: string, message: string) => {
+    notifications.show({
+      title: title,
+      message: message,
+    });
+  };
```
Comment (r-aritra) [inline src/features/Login/Login.tsx:41]: Move to utils module
add styles for
Success: color= green, icon = tick
Error: color = red, icon = cross/x
Info: color = blue icon = info
Should be able to close the notification

File: src/features/Login/Login.tsx [hunk 3]
```diff
@@ -33,21 +33,41 @@ export default function Login() {
...
+
+  const loginMutation = useLogin();
+
   const handleSubmit = async () => {
```
Comment (r-aritra) [inline src/features/Login/Login.tsx:45]: remove async not required

File: src/features/Login/Login.tsx [hunk 4]
```diff
@@ -33,21 +33,41 @@ export default function Login() {
...
-    setVisible(true);
     const data = form.values;
     console.log(data);
```
Comment (r-aritra) [inline src/features/Login/Login.tsx:47]: remove logs

File: src/features/Login/Login.tsx [hunk 5]
```diff
@@ -33,21 +33,41 @@ export default function Login() {
...
-    navigate('/');
-    setVisible(false);
+
+    loginMutation.mutate(
+      {
+        data: { ...data, device_id: '1234321', device_type: 'web' },
```
Comment (r-aritra) [inline src/features/Login/Login.tsx:51]: check if device_id, device_type can be filled dynamically

File: src/features/Login/Login.tsx [hunk 6]
```diff
@@ -33,21 +33,41 @@ export default function Login() {
...
+      },
+      {
+        onSuccess: (responseData) => {
+          console.log('Login successful:', responseData);
+          navigate('/');
```
Comment (r-aritra) [inline src/features/Login/Login.tsx:56]: store token on login success

File: src/test/mocks/browser.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import { getOutdoorEventBookingAPIMock } from '../../models/api.msw';
+
+import registerUser from './json/registerUser.json';
+
+import { setupWorker } from 'msw/browser';
+import { HttpResponse, delay, http } from 'msw';
+
+const mockHandlers = [
+  http.post('*/v1/api/register-user', async () => {
```
Comment (r-aritra) [inline src/test/mocks/browser.ts:9]: check if rest is replaced with http for msw

File: src/features/Login/Login.tsx [hunk 8]
```diff
@@ -33,21 +33,41 @@ export default function Login() {
...
+    );
   };
 
   return (
     <Flex justify="center" align="center" mih={'100vh'}>
-      <LoadingOverlay
-        visible={visible}
-        zIndex={1000}
-        overlayProps={{ radius: 'sm', blur: 2 }}
-      />
```
Comment (r-aritra) [inline src/features/Login/Login.tsx:50]: integrate loader
Check how to integrate loader globally, as fallback using Suspense

File: src/features/Signup/Signup.tsx [hunk 9]
```diff
@@ -36,31 +35,83 @@ export default function Signup() {
     },
   });
 
+  const showNotification = (title: string, message: string) => {
+    notifications.show({
+      title: title,
+      message: message,
+    });
+  };
```
Comment (r-aritra) [inline src/features/Signup/Signup.tsx:43]: same as comment in Login component

File: src/features/Signup/Signup.tsx [hunk 10]
```diff
@@ -36,31 +35,83 @@ export default function Signup() {
...
+    );
   };
 
   return (
     <Flex justify="center" align="center" mih={'100vh'}>
-      <LoadingOverlay
-        visible={visible}
-        zIndex={1000}
-        overlayProps={{ radius: 'sm', blur: 2 }}
-      />
```
Comment (r-aritra) [inline src/features/Signup/Signup.tsx:63]: integrate loader

**Review by r-aritra — CHANGES_REQUESTED** [review]

File: src/features/Login/Login.tsx (hunk 3 above)
Comment (rutvik-balar) [reply src/features/Login/Login.tsx:45]: resolved

File: src/features/Login/Login.tsx (hunk 4 above)
Comment (rutvik-balar) [reply src/features/Login/Login.tsx:47]: resolved

File: src/features/Login/Login.tsx (hunk 5 above)
Comment (rutvik-balar) [reply src/features/Login/Login.tsx:51]: resolved

File: src/features/Signup/Signup.tsx (hunk 9 above)
Comment (rutvik-balar) [reply src/features/Signup/Signup.tsx:43]: resolved

File: src/features/Signup/Signup.tsx (hunk 10 above)
Comment (rutvik-balar) [reply src/features/Signup/Signup.tsx:63]: resolved

File: src/test/mocks/browser.ts (hunk 7 above)
Comment (rutvik-balar) [reply src/test/mocks/browser.ts:9]: resolved

**Review by r-aritra — APPROVED** [review]
lgtm

**Pull request merged**